src/transformers/modeling_tf_utils.py

```diff
@@ -1087,14 +1087,13 @@ def _generate_beam_search(
                 for beam_token_rank, (beam_token_id, beam_token_score) in enumerate(
                     zip(next_tokens[batch_idx], next_scores[batch_idx])
                 ):
-
                     # get beam and token IDs
                     beam_id = beam_token_id // vocab_size
                     token_id = beam_token_id % vocab_size
 
                     effective_beam_id = batch_idx * num_beams + beam_id
                     # add to generated hypotheses if end of sentence or last iteration
-                    if eos_token_id is not None and token_id.numpy() is eos_token_id:
+                    if (eos_token_id is not None) and (token_id.numpy() == eos_token_id):
                         # if beam_token does not belong to top num_beams tokens, it should not be added
```

Comment:
That was a rather big bug (should never use is for int/float comparisons in Python!!! - was probably me who did this though...). This bug led to weird outputs in TF beam search. Before this bug, TF beam search never ended before max_length and therefore had weird results.

Reply:
Good catch @patrickvonplaten